clang/lib/CodeGen/CGExprAgg.cpp

```diff
@@ -536,6 +536,8 @@ void AggExprEmitter::EmitArrayInit(Address DestPtr, llvm::ArrayType *AType,
     CodeGen::CodeGenModule &CGM = CGF.CGM;
     ConstantEmitter Emitter(CGF);
     LangAS AS = ArrayQTy.getAddressSpace();
+    if (CGF.getLangOpts().OpenCL)
+      AS = LangAS::opencl_constant;
```

Comment:
Why is this OpenCL specific?  At first glance, this should be independent of language dialect; the most efficient address-space to store constants depends on the target.

Reply:
That's true as far as it goes, but I'd prefer to make this a bit more general...

It looks like there's an existing GetGlobalConstantAddressSpace() which computes what we want here?

Reply:
Thanks for the suggestion; updated to use `GetGlobalConstantAddressSpace()`.